sway/commands/bar/bindsym.c

```diff
@@ -1,11 +1,69 @@
+#define _XOPEN_SOURCE 500
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include "sway/commands.h"
 #include "sway/config.h"
 #include "list.h"
 #include "log.h"
 #include "stringop.h"
 
 struct cmd_results *bar_cmd_bindsym(int argc, char **argv) {
-	return cmd_results_new(CMD_FAILURE, "bindsym", "TODO"); // TODO
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "bar bindsym", EXPECTED_MORE_THAN, 1))) {
+		return error;
+	}
+	if (!config->current_bar) {
+		return cmd_results_new(CMD_FAILURE, "bar bindsym", "No bar defined.");
+	}
+
+	struct bar_binding *binding = calloc(1, sizeof(struct bar_binding));
+	if (!binding) {
+		return cmd_results_new(CMD_FAILURE, "bar bindsym",
+				"Unable to allocate bar binding");
+	}
+
+	binding->release = false;
```

Comment:
I think this is technically unnecessary due to `calloc` always initialising it to false, but I guess it doesn't hurt to be explicit.